pkgs/development/python-modules/psutil/default.nix

```diff
@@ -18,15 +18,20 @@ buildPythonPackage rec {
   # arch doesn't report frequency is the same way
   # tests segfaults on darwin https://github.com/giampaolo/psutil/issues/1715
   doCheck = !stdenv.isDarwin && stdenv.isx86_64;
-  checkInputs = [ pytest ]
+  checkInputs = [ pytestCheckHook ]
     ++ lib.optionals isPy27 [ mock ipaddress unittest2 ];
-  # out must be referenced as test import paths are relative
+  pytestFlagsArray = [
+    "$out/${python.sitePackages}/psutil/tests/test_system.py"
```

Comment:
Why are we not using the relative path in source?

Reply:
Running tests from build directory doesn't work:
```
_________________ ERROR collecting psutil/tests/test_system.py _________________
ImportError while importing test module '/private/tmp/nix-build-python3.8-psutil-5.8.0.drv-0/psutil-5.8.0/psutil/tests/test_system.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/nix/store/0mz2jkgyjlkwpw21idkyjkibv4xpqnry-python3-3.8.8/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
psutil/__init__.py:121: in <module>
    from . import _psosx as _psplatform
psutil/_psosx.py:15: in <module>
    from . import _psutil_osx as cext
E   ImportError: cannot import name '_psutil_osx' from partially initialized module 'psutil' (most likely due to a circular import) (/private/tmp/nix-build-python3.8-psutil-5.8.0.drv-0/psutil-5.8.0/psutil/__init__.py)
```

Reply:
This was done before. I thought this was new.